code-quality/validation/README.md

```diff
@@ -0,0 +1,83 @@
+<!-- Last updated 9-25-19 -->
+
+# Input Validation 
+
+Input validation is is your first line of defense when creating a secure application.  
+
+## Whitelist or Blacklist? 
+
+The first concept of good input validation is whitelisting versus blacklisting. Whitelist, or inclusive validation, defines a set of valid characters while blacklist, or exclusive validation, defines a set of invalid characters to try to remove. 
+
+Input validation using blacklisting is generally more challenging to implement so it is better practice to perform validation using whitelisting. A great way of defining a whitelist for input validation is to leverage Regular Expressions(RegEx). 
+
+## RegEx Password Validation Examples (JavaScript)
+
+- Mininum length of any 6 chars but newline: 
+```javascript
+/^.{6,}$/
+```
+
+- Check for at least one uppercase character: 
+```javascript
+/([A-Z]+)/g
+```
+
+- Check for at least one lowercase character:
+```javascript
+/([a-z]+)/g
+```
+
+- Check for at least one numeric character: 
+```javascript
+/([0-9]+)/g
+```
+
+
+- Check for at least one special character:
+```javascript
+/([!"#$%&'()*+,\-./:;<=>?@[\]^_`{|}~])+/g
+```
+
+## RegEx Email Validation Example (JavaScript)
+
+- An email is a string (a subset of ASCII characters) separated into two parts by @ symbol, a "personal_info" and a domain, that is personal_info@domain. The length of the personal_info part may be up to 64 characters long and domain name may be up to 253 characters.
+
+```javascript
+/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/
```

Comment:
A few things:

* TLDs can be more than 3 characters now: https://en.wikipedia.org/wiki/List_of_Internet_top-level_domains#ICANN-era_generic_top-level_domains.
* I would make the groups non-capturing groups `(?:[\.-]?\w+)*` just to prevent someone accidentally thinking that they could grab user/domain/etc from the regex.

Reply:
You're right, thanks for the feedback. I'll change to up to 24 characters as that's the longest TDL currently in existence. The longest in English is 14 characters (.cancerresearch).

I'll also change to the groups to non-capturing groups.